util/helpfulIO.py

```diff
@@ -0,0 +1,120 @@
+from typing import Callable
+from ctre import WPI_TalonFX, ControlMode
+from wpilib import RobotBase, PWMVictorSPX, DigitalInput
+from wpimath.controller import PIDController
+from wpimath.system.plant import DCMotor
+from util.ctrecheck import ctreCheckError
+
+import constants
+
+LimitSwitch = Callable[[], bool]
+
+
+class Falcon:  # represents either a simulated motor or a real Falcon 500
+    def __init__(
+        self,
+        name: str,
+        realId: int,
+        simId: int,
+        PGain: int = 1,
+        IGain: int = 0,
+        DGain: int = 0,
+        PIDSlot: int = 0,
+    ) -> None:  # depending on if its in simulation or the real motor, different motors will be used
+        self.name = name
+        self.motor = WPI_TalonFX(realId) if RobotBase.isReal() else PWMVictorSPX(simId)
+
+        if RobotBase.isReal():
+            print(f"Initializing Falcon: {self.name}")
+            if not ctreCheckError(
+                "configFactoryDefault",
+                self.motor.configFactoryDefault(constants.kConfigurationTimeoutLimit),
+            ):
+                return
+            if not ctreCheckError(
+                "config_kP",
+                self.motor.config_kP(
+                    PIDSlot,
+                    PGain,
+                    constants.kConfigurationTimeoutLimit,
+                ),
+            ):
+                return
+            if not ctreCheckError(
+                "config_kI",
+                self.motor.config_kP(
+                    PIDSlot,
+                    IGain,
+                    constants.kConfigurationTimeoutLimit,
+                ),
+            ):
+                return
+            if not ctreCheckError(
+                "config.kD",
+                self.motor.config_kD(
+                    PIDSlot,
+                    DGain,
+                    constants.kConfigurationTimeoutLimit,
+                ),
+            ):
+                return
+            print("Falcon Initialization Complete")
+        else:
+            self.pidController = PIDController(PGain, IGain, DGain)
+            self.simEncoder = 0
+
+    def _setSimMotor(self, amount: float) -> None:
```

Comment:
Might want to add a comment that this value represents percent of motor output and should be in the range [-1, 1]. Probably also want to clamp the input so we don't have the PID controller (or any other input) making the motor spin faster than the free-speed